app/parse.py

```diff
@@ -8,8 +15,68 @@ class Quote:
     tags: list[str]
 
 
+QUOTE_FIELDS = [field.name for field in fields(Quote)]
+
+
+def get_page_html(url: str) -> BeautifulSoup:
+    response = requests.get(url)
+    response.encoding = "utf-8"
+    try:
+        return BeautifulSoup(response.text, "html.parser")
+    except UnicodeDecodeError:
+        print("Error parsing the following content from URL:", url)
+        print(response.text)
+        raise
+
+
+def get_path_to_next_page(url: str) -> Union[str, bool]:
+    link_to_next_page = get_page_html(url).select_one(
+        ".pager > .next")
+    if link_to_next_page:
+        return str(link_to_next_page.find_next("a")["href"])[1:]
+    return False
+
+
+def get_quote_from_single_page(url: str) -> list[Quote]:
+    current_url = url
+    quotes_div = get_page_html(current_url).select(".quote")
+    quotes_text = [text.select_one(".text").text for text in quotes_div]
+    quotes_author = [
+        text.select_one(".author").text for text in quotes_div
+    ]
+    quotes_tags = [
+        [tag.text for tag in text.select(".tag")] for text in quotes_div
+    ]
+    return [
+        Quote(
+            text=quotes_text[index],
+            author=quotes_author[index],
+            tags=quotes_tags[index]
+        ) for index in range(len(quotes_div))
+    ]
+
+
+def get_all_quotes(url: str) -> list[Quote]:
+    current_url = url
+
+    quotes = get_quote_from_single_page(current_url)
+    page = 1
+
+    while get_path_to_next_page(current_url):
+        page += 1
+        current_url = url + f"page/{page}/"
+        quotes += get_quote_from_single_page(current_url)
+
+    return quotes
+
+
```

Comment:
I think it's better to create a class QuoteParser and rewrite all these functions as methods of the class